sky/clouds/service_catalog/scp_catalog.py

```diff
@@ -0,0 +1,146 @@
+"""SCP Cloud Catalog.
+
+This module loads the service catalog file and can be used to query
+instance types and pricing information for SCP.
+"""
+
+import typing
+from typing import Dict, List, Optional, Tuple
+
+from sky.clouds.service_catalog import common
+from sky.skylet.providers.scp import scp_utils
+from sky.utils import ux_utils
+
+if typing.TYPE_CHECKING:
+    from sky.clouds import cloud
+
+_df = common.read_catalog('scp/vms.csv')
+_image_df = common.read_catalog('scp/images.csv')
+_DEFAULT_NUM_VCPUS = 8
+_DEFAULT_MEMORY_CPU_RATIO = 2
+
+
+def crop_available_region(df):
+    scp_client = scp_utils.SCPClient()
+    service_zones = scp_client.list_service_zone_names()
+    return df[df['Region'].isin(service_zones)]
+
+
+_df = crop_available_region(_df)
```

Comment:
> 2. It seems adding SCP slows down SkyPilot.
> I guess some SCP http requests are invoked every time SkyPilot is loaded.
> It is right that, when SCP is Enabled, the catalog service invokes an SCP HTTP request. (checked it takes 0.2~0.5 secs)
> An SCP user is allowed to access not all but only designated service zones(=regions) configured in the SCP project.
> So it is necessity to obtain the available service zones, via HTTP requests during the catalog service.
> We hope this feature won't be too much of a problem for you, since it happens only SCP is used.

@dreamersnme Can we avoid the slowdown by caching the data locally and update them less frequently? For example, here the `list_service_zone_names` method is called every time the scp module is loaded. Why don't we cache this data locally?